appointment-service/src/main/java/com/example/AppointmentService/Schedule/Schedule.java

```diff
@@ -31,24 +31,39 @@ public class Schedule {
     private int centerId;
     private int vaccineTypeId;
 
-    @Column(name = "totalSlot")
-    @Formula("(SELECT COUNT(*) FROM APPOINTMENT a WHERE a.schedule_id = schedule_id)")
-    private int totalSlot;
+//    @Column(name = "totalSlot")
+//    private int totalSlot;
 
-    @Formula("(SELECT COUNT(*) FROM APPOINTMENT a WHERE a.schedule_id = schedule_id AND a.status = 'open')")
     private int openSlot;
-    @Formula("(SELECT COUNT(*) FROM APPOINTMENT a WHERE a.schedule_id = schedule_id AND a.status = 'completed')")
     private int completedSlot;
     @JsonManagedReference
     @NonNull
     @OneToMany(mappedBy = "schedule",fetch = FetchType.LAZY,cascade = CascadeType.ALL)
     private List<Appointment> appointmentList = new ArrayList<>();
 
-    public Schedule(Date date, Time time,int centerId, int vaccineTypeId) {
+    public Schedule(Date date, Time time,int centerId, int vaccineTypeId, int openSlot) {
         this.date = date;
         this.centerId = centerId;
         this.vaccineTypeId = vaccineTypeId;
         this.time = time;
-
+        this.openSlot = openSlot;
```

Comment:
What is totalSlot?  Are totalSlot and openSlot the same at the ctor?  totalSlot should be fixed after object created and openSlot keeps track of available shot for appointment.  Each appointment should use one openSlot.